### DISCUSSION TITLE
BlockRecords["MyBlock"].Clone();

### DISCUSSION THREAD

**Discussion by CCvallecano** [DomCR/ACadSharp#453] [General]
Good afternoon, I have inserted your library with a nuget package. I am trying to insert a block from an external file with this code:
 const string _file = "../../../LibraryBlock/Block/133499.dwg";
 CadDocument docToRead;
 BlockRecord block;
 CadDocument transfer = new CadDocument();
 using (DwgReader reader = new DwgReader(_file))
 {
     docToRead = reader.Read();

 }

 block = (BlockRecord)docToRead.BlockRecords["MyBlock"].Clone();
 transfer.BlockRecords.Add(block);

But it gives me this error

 System.Collections.Generic.KeyNotFoundException: 'The given key was not present in the dictionary.'

I saw that there was a bug already fixed

th
Juan

**Comment by DomCR** [comment]
Hi @CCvallecano,

I don't quite understand the question, but seeing at the exception I think that the issue is that the block with the name "MyBlock" doesn't exist in the file, can you confirm that?

**Reply by CCvallecano** [reply]
Most likely it's me
I want to create a .dwg (Mydwg.dwg) and insert into the other .dwg ("../../../LibraryBlock/Block/133499.dwg") at a certain point and with a rotation

**Comment by CCvallecano** [comment]
BlockRecord record2 = new BlockRecord("100009x-2");
CreateDoc.BlockRecords.Add(record2);
const string _file = "../../../LibraryBlock/Block/133499.dwg";
CadDocument docToRead;
BlockRecord block;
CadDocument transfer = new CadDocument();
using (DwgReader reader = new DwgReader(_file))
{
    docToRead = reader.Read();

}

block = (BlockRecord)docToRead.BlockRecords["100009x-2"].Clone();
transfer.BlockRecords.Add(block);

I created the block but I have the same problem

**Reply by DomCR** [reply]
You've created the block in another document, `CreateDoc`, the one that you are reading doesn't seem to contain the block "100009x-2".

You can use the method `BlockRecords.TryGetValue()` to check if the Table contains that block without triggering the exception.

Also it may help you to log all the blocks in the table to see what the file contains.

**Comment by CCvallecano** [comment]
Thank you very much, I understand it now and it works. I thought we could insert a dwg into our new drawing. I will create the block inside another block. I love the library. Thank you very much.

**Comment by CCvallecano** [comment]
One more question, the 3D solids in the cloned block are not cloned. Am I doing something wrong?

**Reply by DomCR** [reply]
3D solids are not implemented yet, here is the PR meant to fix it:

- https://github.com/DomCR/ACadSharp/pull/191

**Reply by CCvallecano** [reply]
th
